### DISCUSSION TITLE
Autologin

### DISCUSSION THREAD

**Discussion by bbakirtas** [InnovateAsterisk/Browser-Phone#389] [Q&A] [answered]
Greetings, I want the user to automatically log in with the account I have specified. I have edited the phone.js file, but it did not work.
my port 7443

> let profileName = 'Customer';       // eg: Keyla James
> let wssServer = 'myserver';           // eg: raspberrypi.local
> let WebSocketPort = '7443';   // eg: 444 | 4443
> let ServerPath = '/ws';         // eg: /ws
> let SipDomain = 'myserver';           // eg: raspberrypi.local
> let SipUsername = '102';       // eg: webrtc
> let SipPassword = '102abc';       // eg: webrtc

**Comment by InnovateAsterisk** [comment, marked as answer]
I wouldn’t suggest making changes to the phone.js file. Sorry, it should be more clear. Just load another .js file before, and add any of the variable definitions to that file as phoneOptions. 

Take a look at the current index.html file. You should see that the phone.js file loads last, so like the sample provided, just define the phoneOptions object, and this will provision the phone.

The system makes extensive use of the local storage, so changing the variables in the phone.js also breaks this.

**Reply by bbakirtas** [reply]
I did as you suggested. but I couldn't run it.Profile name comes up but can't register. 
 
var phoneOptions = {
            	wssServer : "server.com",
                WebSocketPort : "7443",
                ServerPath : "/ws",
                SipDomain : "server.com",
                SipUsername : "102",
                SipPassword : "102abc",
                profileName : "102 Test",  
                loadAlternateLang : true        
            }

**Comment by InnovateAsterisk** [comment]
What appears in the console log? Some error?

**Reply by bbakirtas** [reply]
it does not give any error in console

**Reply by bbakirtas** [reply]
other setting are working fine

EnableAccountSettings:1,
EnableAppearanceSettings:0,
EnableNotificationSettings:1

**Comment by InnovateAsterisk** [comment]
Ok, but what isn’t working? You are expecting an register, right? Can you check the sip messages. (Find the ws connection in the network tab, select it, then select the messages tab inside that section. There you see all the sip messages.)

**Reply by bbakirtas** [reply]
I can't register to the system with my phone account. (102) You have to fill out the form and register accordingly.

**Reply by bbakirtas** [reply]
so these fields that we do in phoneOptions do not work
wssServer : "server.com",
WebSocketPort: "7443",
ServerPath : "/ws",
SipDomain: "server.com",
SipUsername: "102",
SipPassword: "102abc",

**Reply by InnovateAsterisk** [reply]
Please make sure that you are using the unmodified phone.js file. Then, there is another field called "profileUserID". This can be populated from the phoneOptions, but it must be the same each time. Its the identity for the local storage, and based off this ID, the entire configuration of a user exists. 

Technically... although a bit ugly... you can have multiple accounts load onto the same browser instance, by setting this value. As in, Paul (ID X), can log in, then log out, and Peter (ID Y) can log in, and they will have partitioned data.

What's important here, is that you set it, and the next visit, you provide the same ID again. The phone.js code will make this code up the first time, and then looks for it on subsequent loads. This is why you appear to need to apply the setting each time even though the fields are provisioned. So by providing the same ```profileUserID``` each time, you are basically handling that process your self.

Also note that if you are provisioning settings, you may want to disable the ability for users to change them with ```EnableAccountSettings```.

**Reply by bbakirtas** [reply]
Also note that if you are provisioning settings, you may want to disable the ability for users to change them with EnableAccountSettings. **correct**

wssServer : 'server',
  WebSocketPort : '7443',
  ServerPath : '/ws',
  SipDomain : 'server',
  SipUsername : '103',
  SipPassword : '103abc',
  profileName : '103', 
  profileUser : '103', 
  EnableAccountSettings:0,
  EnableAppearanceSettings:1,
  EnableNotificationSettings:1,
  loadAlternateLang : true   

now its worked :) thanks alot 
I was expecting automatic login.
logged in after clicking the button.
